spec/requests/checkouts_spec.rb

```diff
@@ -118,4 +118,52 @@
       expect(response).to redirect_to(checkouts_url)
     end
   end
+
+  describe "PATCH /return" do
+    it "updates the return time of requested checkout" do
+      patch return_checkout_url(checkout)
+      checkout.reload
+      expect(checkout.return_time).to be <= DateTime.current
+    end
+    it "redirects to the checkouts list" do
+      patch return_checkout_url(checkout)
+      expect(response).to redirect_to(checkouts_url)
+    end
+  end
+
+  describe "PATCH /return_all" do
+    before :each do
+      FactoryBot.reload
+      FactoryBot.create_list(:checkout, 2)
+      FactoryBot.create(:checkout, user: user)
+    end
+
+    context "as a regular user" do
+      it "updates the user's active checkouts" do
+        patch return_all_checkouts_url
+        expect(Checkout.where(user_id: user.id).first.return_time).to be <= DateTime.current
+      end
+      it "does not update other checkouts" do
+        patch return_all_checkouts_url
+        expect(Checkout.where.not(user_id: user.id).first.return_time).to be > DateTime.current
+        expect(Checkout.where.not(user_id: user.id).second.return_time).to be > DateTime.current
+      end
+      it "redirects to the checkouts list" do
+        patch return_all_checkouts_url
+        expect(response).to redirect_to(checkouts_url)
+      end
+    end
+    context "as an admin user" do
+      let(:user) { FactoryBot.create(:admin) }
+      it "updates all active checkouts" do
+        patch return_all_checkouts_url
+        sleep 1
```

Comment:
When I was running the tests on my system, the test would fail because the check would occur before the database had finished updating so I would receive '0' instead of '3'. There may be a better way than using the sleep here, but it was what I was able to come up with to get the test working.

Reply:
Cool, I'll see if I can come up with something better.

Reply:
I think I found it.  I think the issue was that `DateTime.to_s` drops the milliseconds.  Passing `DateTime.current` to the where command and letting it do the casting fixes it for me.
```
expect(Checkout.where("return_time <= ?", DateTime.current).count).to eq(3)
```

Reply:
That makes sense. Thanks!